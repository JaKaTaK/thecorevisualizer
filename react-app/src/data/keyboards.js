const keyboards = {
		'USQwerty': [
			{
				'ESC': 'ESC',
				'_0': '_0',
				'F1': 'F1',
				'F2': 'F2',
				'F3': 'F3',
				'F4': 'F4',
				'_1': 0.5,
				'F5': 'F5',
				'F6': 'F6',
				'F7': 'F7',
				'F8': 'F8',
				'_2': 0.5,
				'F9': 'F9',
				'F10': 'F10',
				'F11': 'F11',
				'F12': 'F12'
			},
			{},
			{
				'`': '`',
				'1 ': '1 ',
				'2 ': '2 ',
				'3 ': '3 ',
				'4 ': '4 ',
				'5 ': '5 ',
				'6 ': '6 ',
				'7 ': '7 ',
				'8 ': '8 ',
				'9 ': '9 ',
				'0 ': '0 ',
				'-': '-',
				'=': '=',
				'BACKSPACE': 2
			},
			{
				'TAB': 1.5,
				'Q': 'Q',
				'W': 'W',
				'E': 'E',
				'R': 'R',
				'T': 'T',
				'Y': 'Y',
				'U': 'U',
				'I': 'I',
				'O': 'O',
				'P': 'P',
				'[': '[',
				']': ']',
				'\\': 1.5
			},
			{
				'CAPS': 1.75,
				'A': 'A',
				'S': 'S',
				'D': 'D',
				'F': 'F',
				'G': 'G',
				'H': 'H',
				'J': 'J',
				'K': 'K',
				'L': 'L',
				';': ';',
				'\'': '\'',
				'ENTER': 2.25
			},
			{
				'SHIFT L': 2.25,
				'Z': 'Z',
				'X': 'X',
				'C': 'C',
				'V': 'V',
				'B': 'B',
				'N': 'N',
				'M': 'M',
				',': ',',
				'.': '.',
				'/': '/',
				'SHIFT R': 2.75
			},
			{
				'CTRL L': 1.5,
				'WIN L': 'WIN L',
				'ALT L': 1.5,
				'SPACE': 6,
				'ALT R': 1.5,
				'WIN R': 'WIN R',
				'MENU': 'MENU',
				'CTRL R': 1.5
			}
		],
		'USDvorak': [
			{
				'ESC': 'ESC',
				'_0': '_0',
				'F1': 'F1',
				'F2': 'F2',
				'F3': 'F3',
				'F4': 'F4',
				'_1': 0.5,
				'F5': 'F5',
				'F6': 'F6',
				'F7': 'F7',
				'F8': 'F8',
				'_2': 0.5,
				'F9': 'F9',
				'F10': 'F10',
				'F11': 'F11',
				'F12': 'F12'
			},
			{},
			{
				'`': '`',
				'1 ': '1 ',
				'2 ': '2 ',
				'3 ': '3 ',
				'4 ': '4 ',
				'5 ': '5 ',
				'6 ': '6 ',
				'7 ': '7 ',
				'8 ': '8 ',
				'9 ': '9 ',
				'0 ': '0 ',
				'[': '[',
				']': ']',
				'BACKSPACE': 2
			},
			{
				'TAB': 1.5,
				'\'': '\'',
				',': ',',
				'.': '.',
				'P': 'P',
				'Y': 'Y',
				'F': 'F',
				'G': 'G',
				'C': 'C',
				'R': 'R',
				'L': 'L',
				'/': '/',
				'=': '=',
				'\\': 1.5
			},
			{
				'CAPS': 1.75,
				'A': 'A',
				'O': 'O',
				'E': 'E',
				'U': 'U',
				'I': 'I',
				'D': 'D',
				'H': 'H',
				'T': 'T',
				'N': 'N',
				'S': 'S',
				'-': '-',
				'ENTER': 2.25
			},
			{
				'SHIFT L': 2.25,
				';': ';',
				'Q': 'Q',
				'J': 'J',
				'K': 'K',
				'X': 'X',
				'B': 'B',
				'M': 'M',
				'W': 'W',
				'V': 'V',
				'Z': 'Z',
				'SHIFT R': 2.75
			},
			{
				'CTRL L': 1.5,
				'WIN L': 'WIN L',
				'ALT L': 1.5,
				'SPACE': 6,
				'ALT R': 1.5,
				'WIN R': 'WIN R',
				'MENU': 'MENU',
				'CTRL R': 1.5
			}
		],
		'FrenchAzerty': [
			{
				'ESC': 'ESC',
				'_0': '_0',
				'F1': 'F1',
				'F2': 'F2',
				'F3': 'F3',
				'F4': 'F4',
				'_1': 0.5,
				'F5': 'F5',
				'F6': 'F6',
				'F7': 'F7',
				'F8': 'F8',
				'_2': 0.5,
				'F9': 'F9',
				'F10': 'F10',
				'F11': 'F11',
				'F12': 'F12'
			},
			{},
			{
				'<': '<',
				'1 ': '1 ',
				'2 ': '2 ',
				'3 ': '3 ',
				'4 ': '4 ',
				'5 ': '5 ',
				'6 ': '6 ',
				'7 ': '7 ',
				'8 ': '8 ',
				'9 ': '9 ',
				'0 ': '0 ',
				'-': '-',
				'=': '=',
				'BACKSPACE': 2
			},
			{
				'TAB': 1.5,
				'A': 'A',
				'Z': 'Z',
				'E': 'E',
				'R': 'R',
				'T': 'T',
				'Y': 'Y',
				'U': 'U',
				'I': 'I',
				'O': 'O',
				'P': 'P',
				'[': '[',
				']': ']',
				'\\': 1.5
			},
			{
				'CAPS': 1.75,
				'Q': 'Q',
				'S': 'S',
				'D': 'D',
				'F': 'F',
				'G': 'G',
				'H': 'H',
				'J': 'J',
				'K': 'K',
				'L': 'L',
				'M': 'M',
				'`': '`',
				'ENTER': 2.25
			},
			{
				'SHIFT L': 2.25,
				'W': 'W',
				'X': 'X',
				'C': 'C',
				'V': 'V',
				'B': 'B',
				'N': 'N',
				',': ',',
				';': ';',
				':': ':',
				'=': '=',
				'SHIFT R': 2.75
			},
			{
				'CTRL L': 1.5,
				'WIN L': 'WIN L',
				'ALT L': 1.5,
				'SPACE': 6,
				'ALT R': 1.5,
				'WIN R': 'WIN R',
				'MENU': 'MENU',
				'CTRL R': 1.5
			}
		],
		'German': [
			{
				'ESC': 'ESC',
				'_0': '_0',
				'F1': 'F1',
				'F2': 'F2',
				'F3': 'F3',
				'F4': 'F4',
				'_1': 0.5,
				'F5': 'F5',
				'F6': 'F6',
				'F7': 'F7',
				'F8': 'F8',
				'_2': 0.5,
				'F9': 'F9',
				'F10': 'F10',
				'F11': 'F11',
				'F12': 'F12'
			},
			{},
			{
				'<': '<',
				'1 ': '1 ',
				'2 ': '2 ',
				'3 ': '3 ',
				'4 ': '4 ',
				'5 ': '5 ',
				'6 ': '6 ',
				'7 ': '7 ',
				'8 ': '8 ',
				'9 ': '9 ',
				'0 ': '0 ',
				'ß': 'ß',
				'´': '´',
				'BACKSPACE': 2
			},
			{
				'TAB': 1.5,
				'Q': 'Q',
				'W': 'W',
				'E': 'E',
				'R': 'R',
				'T': 'T',
				'Z': 'Z',
				'U': 'U',
				'I': 'I',
				'O': 'O',
				'P': 'P',
				'ü': 'ü',
				'+': '+',
				'\\': 1.5
			},
			{
				'CAPS': 1.75,
				'A': 'A',
				'S': 'S',
				'D': 'D',
				'F': 'F',
				'G': 'G',
				'H': 'H',
				'J': 'J',
				'K': 'K',
				'L': 'L',
				'ö': 'ö',
				'ä': 'ä',
				'ENTER': 2.25
			},
			{
				'SHIFT L': 2.25,
				'Y': 'Y',
				'X': 'X',
				'C': 'C',
				'V': 'V',
				'B': 'B',
				'N': 'N',
				'M': 'M',
				',': ',',
				'.': '.',
				'-': '-',
				'SHIFT R': 2.75
			},
			{
				'CTRL L': 1.5,
				'WIN L': 'WIN L',
				'ALT L': 1.5,
				'SPACE': 6,
				'ALT R': 1.5,
				'WIN R': 'WIN R',
				'MENU': 'MENU',
				'CTRL R': 1.5
			}
		],
		'USColemak': [
			{
				'ESC': 'ESC',
				'_0': '_0',
				'F1': 'F1',
				'F2': 'F2',
				'F3': 'F3',
				'F4': 'F4',
				'_1': 0.5,
				'F5': 'F5',
				'F6': 'F6',
				'F7': 'F7',
				'F8': 'F8',
				'_2': 0.5,
				'F9': 'F9',
				'F10': 'F10',
				'F11': 'F11',
				'F12': 'F12'
			},
			{},
			{
				'`': '`',
				'1 ': '1 ',
				'2 ': '2 ',
				'3 ': '3 ',
				'4 ': '4 ',
				'5 ': '5 ',
				'6 ': '6 ',
				'7 ': '7 ',
				'8 ': '8 ',
				'9 ': '9 ',
				'0 ': '0 ',
				'-': '-',
				'=': '=',
				'BACKSPACE': 2
			},
			{
				'TAB': 1.5,
				'Q': 'Q',
				'W': 'W',
				'F': 'F',
				'P': 'P',
				'G': 'G',
				'J': 'J',
				'L': 'L',
				'U': 'U',
				'Y': 'Y',
				';': ';',
				'[': '[',
				']': ']',
				'\\': 1.5
			},
			{
				'CAPS': 1.75,
				'A': 'A',
				'R': 'R',
				'S': 'S',
				'T': 'T',
				'D': 'D',
				'H': 'H',
				'N': 'N',
				'E': 'E',
				'I': 'I',
				'O': 'O',
				'\'': '\'',
				'ENTER': 2.25
			},
			{
				'SHIFT L': 2.25,
				'Z': 'Z',
				'X': 'X',
				'C': 'C',
				'V': 'V',
				'B': 'B',
				'K': 'K',
				'M': 'M',
				',': ',',
				'.': '.',
				'/': '/',
				'SHIFT R': 2.75
			},
			{
				'CTRL L': 1.5,
				'WIN L': 'WIN L',
				'ALT L': 1.5,
				'SPACE': 6,
				'ALT R': 1.5,
				'WIN R': 'WIN R',
				'MENU': 'MENU',
				'CTRL R': 1.5
			}
		]
	};
export { keyboards };
