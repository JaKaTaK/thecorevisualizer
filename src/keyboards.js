// Generated by CoffeeScript 2.2.1
(function() {
  var i, j, k, key, keyCodeMaps, keyboard, keyboards, len, len1, row, size;

  exports.keyboards = keyboards = {
    'USQwerty': [
      {
        'ESC': 'ESC',
        '_0': '_0',
        'F1': 'F1',
        'F2': 'F2',
        'F3': 'F3',
        'F4': 'F4',
        '_1': 0.5,
        'F5': 'F5',
        'F6': 'F6',
        'F7': 'F7',
        'F8': 'F8',
        '_2': 0.5,
        'F9': 'F9',
        'F10': 'F10',
        'F11': 'F11',
        'F12': 'F12'
      },
      {},
      {
        '`': '`',
        '1 ': '1 ',
        '2 ': '2 ',
        '3 ': '3 ',
        '4 ': '4 ',
        '5 ': '5 ',
        '6 ': '6 ',
        '7 ': '7 ',
        '8 ': '8 ',
        '9 ': '9 ',
        '0 ': '0 ',
        '-': '-',
        '=': '=',
        'BACKSPACE': 2
      },
      {
        'TAB': 1.5,
        'Q': 'Q',
        'W': 'W',
        'E': 'E',
        'R': 'R',
        'T': 'T',
        'Y': 'Y',
        'U': 'U',
        'I': 'I',
        'O': 'O',
        'P': 'P',
        '[': '[',
        ']': ']',
        '\\': 1.5
      },
      {
        'CAPS': 1.75,
        'A': 'A',
        'S': 'S',
        'D': 'D',
        'F': 'F',
        'G': 'G',
        'H': 'H',
        'J': 'J',
        'K': 'K',
        'L': 'L',
        ';': ';',
        '\'': '\'',
        'ENTER': 2.25
      },
      {
        'SHIFT L': 2.25,
        'Z': 'Z',
        'X': 'X',
        'C': 'C',
        'V': 'V',
        'B': 'B',
        'N': 'N',
        'M': 'M',
        ',': ',',
        '.': '.',
        '/': '/',
        'SHIFT R': 2.75
      },
      {
        'CTRL L': 1.5,
        'WIN L': 'WIN L',
        'ALT L': 1.5,
        'SPACE': 6,
        'ALT R': 1.5,
        'WIN R': 'WIN R',
        'MENU': 'MENU',
        'CTRL R': 1.5
      }
    ],
    'USDvorak': [
      {
        'ESC': 'ESC',
        '_0': '_0',
        'F1': 'F1',
        'F2': 'F2',
        'F3': 'F3',
        'F4': 'F4',
        '_1': 0.5,
        'F5': 'F5',
        'F6': 'F6',
        'F7': 'F7',
        'F8': 'F8',
        '_2': 0.5,
        'F9': 'F9',
        'F10': 'F10',
        'F11': 'F11',
        'F12': 'F12'
      },
      {},
      {
        '`': '`',
        '1 ': '1 ',
        '2 ': '2 ',
        '3 ': '3 ',
        '4 ': '4 ',
        '5 ': '5 ',
        '6 ': '6 ',
        '7 ': '7 ',
        '8 ': '8 ',
        '9 ': '9 ',
        '0 ': '0 ',
        '[': '[',
        ']': ']',
        'BACKSPACE': 2
      },
      {
        'TAB': 1.5,
        '\'': '\'',
        ',': ',',
        '.': '.',
        'P': 'P',
        'Y': 'Y',
        'F': 'F',
        'G': 'G',
        'C': 'C',
        'R': 'R',
        'L': 'L',
        '/': '/',
        '=': '=',
        '\\': 1.5
      },
      {
        'CAPS': 1.75,
        'A': 'A',
        'O': 'O',
        'E': 'E',
        'U': 'U',
        'I': 'I',
        'D': 'D',
        'H': 'H',
        'T': 'T',
        'N': 'N',
        'S': 'S',
        '-': '-',
        'ENTER': 2.25
      },
      {
        'SHIFT L': 2.25,
        ';': ';',
        'Q': 'Q',
        'J': 'J',
        'K': 'K',
        'X': 'X',
        'B': 'B',
        'M': 'M',
        'W': 'W',
        'V': 'V',
        'Z': 'Z',
        'SHIFT R': 2.75
      },
      {
        'CTRL L': 1.5,
        'WIN L': 'WIN L',
        'ALT L': 1.5,
        'SPACE': 6,
        'ALT R': 1.5,
        'WIN R': 'WIN R',
        'MENU': 'MENU',
        'CTRL R': 1.5
      }
    ],
    'FrenchAzerty': [
      {
        'ESC': 'ESC',
        '_0': '_0',
        'F1': 'F1',
        'F2': 'F2',
        'F3': 'F3',
        'F4': 'F4',
        '_1': 0.5,
        'F5': 'F5',
        'F6': 'F6',
        'F7': 'F7',
        'F8': 'F8',
        '_2': 0.5,
        'F9': 'F9',
        'F10': 'F10',
        'F11': 'F11',
        'F12': 'F12'
      },
      {},
      {
        '<': '<',
        '1 ': '1 ',
        '2 ': '2 ',
        '3 ': '3 ',
        '4 ': '4 ',
        '5 ': '5 ',
        '6 ': '6 ',
        '7 ': '7 ',
        '8 ': '8 ',
        '9 ': '9 ',
        '0 ': '0 ',
        '-': '-',
        '=': '=',
        'BACKSPACE': 2
      },
      {
        'TAB': 1.5,
        'A': 'A',
        'Z': 'Z',
        'E': 'E',
        'R': 'R',
        'T': 'T',
        'Y': 'Y',
        'U': 'U',
        'I': 'I',
        'O': 'O',
        'P': 'P',
        '[': '[',
        ']': ']',
        '\\': 1.5
      },
      {
        'CAPS': 1.75,
        'Q': 'Q',
        'S': 'S',
        'D': 'D',
        'F': 'F',
        'G': 'G',
        'H': 'H',
        'J': 'J',
        'K': 'K',
        'L': 'L',
        'M': 'M',
        '`': '`',
        'ENTER': 2.25
      },
      {
        'SHIFT L': 2.25,
        'W': 'W',
        'X': 'X',
        'C': 'C',
        'V': 'V',
        'B': 'B',
        'N': 'N',
        ',': ',',
        ';': ';',
        ':': ':',
        '=': '=',
        'SHIFT R': 2.75
      },
      {
        'CTRL L': 1.5,
        'WIN L': 'WIN L',
        'ALT L': 1.5,
        'SPACE': 6,
        'ALT R': 1.5,
        'WIN R': 'WIN R',
        'MENU': 'MENU',
        'CTRL R': 1.5
      }
    ],
    'German': [
      {
        'ESC': 'ESC',
        '_0': '_0',
        'F1': 'F1',
        'F2': 'F2',
        'F3': 'F3',
        'F4': 'F4',
        '_1': 0.5,
        'F5': 'F5',
        'F6': 'F6',
        'F7': 'F7',
        'F8': 'F8',
        '_2': 0.5,
        'F9': 'F9',
        'F10': 'F10',
        'F11': 'F11',
        'F12': 'F12'
      },
      {},
      {
        '<': '<',
        '1 ': '1 ',
        '2 ': '2 ',
        '3 ': '3 ',
        '4 ': '4 ',
        '5 ': '5 ',
        '6 ': '6 ',
        '7 ': '7 ',
        '8 ': '8 ',
        '9 ': '9 ',
        '0 ': '0 ',
        'ß': 'ß',
        '´': '´',
        'BACKSPACE': 2
      },
      {
        'TAB': 1.5,
        'Q': 'Q',
        'W': 'W',
        'E': 'E',
        'R': 'R',
        'T': 'T',
        'Z': 'Z',
        'U': 'U',
        'I': 'I',
        'O': 'O',
        'P': 'P',
        'ü': 'ü',
        '+': '+',
        '\\': 1.5
      },
      {
        'CAPS': 1.75,
        'A': 'A',
        'S': 'S',
        'D': 'D',
        'F': 'F',
        'G': 'G',
        'H': 'H',
        'J': 'J',
        'K': 'K',
        'L': 'L',
        'ö': 'ö',
        'ä': 'ä',
        'ENTER': 2.25
      },
      {
        'SHIFT L': 2.25,
        'Y': 'Y',
        'X': 'X',
        'C': 'C',
        'V': 'V',
        'B': 'B',
        'N': 'N',
        'M': 'M',
        ',': ',',
        '.': '.',
        '-': '-',
        'SHIFT R': 2.75
      },
      {
        'CTRL L': 1.5,
        'WIN L': 'WIN L',
        'ALT L': 1.5,
        'SPACE': 6,
        'ALT R': 1.5,
        'WIN R': 'WIN R',
        'MENU': 'MENU',
        'CTRL R': 1.5
      }
    ],
    'USColemak': [
      {
        'ESC': 'ESC',
        '_0': '_0',
        'F1': 'F1',
        'F2': 'F2',
        'F3': 'F3',
        'F4': 'F4',
        '_1': 0.5,
        'F5': 'F5',
        'F6': 'F6',
        'F7': 'F7',
        'F8': 'F8',
        '_2': 0.5,
        'F9': 'F9',
        'F10': 'F10',
        'F11': 'F11',
        'F12': 'F12'
      },
      {},
      {
        '`': '`',
        '1 ': '1 ',
        '2 ': '2 ',
        '3 ': '3 ',
        '4 ': '4 ',
        '5 ': '5 ',
        '6 ': '6 ',
        '7 ': '7 ',
        '8 ': '8 ',
        '9 ': '9 ',
        '0 ': '0 ',
        '-': '-',
        '=': '=',
        'BACKSPACE': 2
      },
      {
        'TAB': 1.5,
        'Q': 'Q',
        'W': 'W',
        'F': 'F',
        'P': 'P',
        'G': 'G',
        'J': 'J',
        'L': 'L',
        'U': 'U',
        'Y': 'Y',
        ';': ';',
        '[': '[',
        ']': ']',
        '\\': 1.5
      },
      {
        'CAPS': 1.75,
        'A': 'A',
        'R': 'R',
        'S': 'S',
        'T': 'T',
        'D': 'D',
        'H': 'H',
        'N': 'N',
        'E': 'E',
        'I': 'I',
        'O': 'O',
        '\'': '\'',
        'ENTER': 2.25
      },
      {
        'SHIFT L': 2.25,
        'Z': 'Z',
        'X': 'X',
        'C': 'C',
        'V': 'V',
        'B': 'B',
        'K': 'K',
        'M': 'M',
        ',': ',',
        '.': '.',
        '/': '/',
        'SHIFT R': 2.75
      },
      {
        'CTRL L': 1.5,
        'WIN L': 'WIN L',
        'ALT L': 1.5,
        'SPACE': 6,
        'ALT R': 1.5,
        'WIN R': 'WIN R',
        'MENU': 'MENU',
        'CTRL R': 1.5
      }
    ]
  };

  exports.keyCodeMaps = keyCodeMaps = {
    'USQwerty': {
      27: 'ESC',
      9: 'TAB',
      16: 'SHIFT',
      17: 'CTRL',
      18: 'ALT',
      32: 'SPACE',
      13: 'ENTER',
      46: 'BACKSPACE',
      "48": "0",
      "49": "1",
      "50": "2",
      "51": "3",
      "52": "4",
      "53": "5",
      "54": "6",
      "55": "7",
      "56": "8",
      "57": "9",
      59: ';',
      "65": "a",
      "66": "b",
      "67": "c",
      "68": "d",
      "69": "e",
      "70": "f",
      "71": "g",
      "72": "h",
      "73": "i",
      "74": "j",
      "75": "k",
      "76": "l",
      "77": "m",
      "78": "n",
      "79": "o",
      "80": "p",
      "81": "q",
      "82": "r",
      "83": "s",
      "84": "t",
      "85": "u",
      "86": "v",
      "87": "w",
      "88": "x",
      "89": "y",
      "90": "z",
      "186": ";",
      "187": "=",
      "188": ",",
      "189": "-",
      "190": ".",
      "191": "/",
      "192": "`",
      "219": "[",
      "220": "\\",
      "221": "]",
      "222": "'",
      112: 'F1',
      113: 'F2',
      114: 'F3',
      115: 'F4',
      116: 'F5',
      117: 'F6',
      118: 'F7',
      119: 'F8',
      120: 'F9',
      121: 'F10',
      122: 'F11',
      123: 'F12'
    }
  };

  for (k in keyboards) {
    keyCodeMaps[k] = keyCodeMaps['USQwerty'];
  }

  for (i = 0, len = keyboards.length; i < len; i++) {
    keyboard = keyboards[i];
    for (j = 0, len1 = keyboard.length; j < len1; j++) {
      row = keyboard[j];
      for (key in row) {
        size = row[key];
        if (typeof size === 'string') {
          row[key] = 1;
        }
      }
    }
  }

}).call(this);
