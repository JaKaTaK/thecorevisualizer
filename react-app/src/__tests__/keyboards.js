import { keyboards }   from '../data/keyboards.js';

test('keyboard strings to be correct are correct for the core 4.0', () => {
	const result = keyboards; 
	expect(result).toMatchSnapshot();
});

