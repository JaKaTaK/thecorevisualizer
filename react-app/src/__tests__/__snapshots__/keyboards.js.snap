// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keyboard strings to be correct are correct for the core 4.0 1`] = `
Object {
  "FrenchAzerty": Array [
    Object {
      "ESC": "ESC",
      "F1": "F1",
      "F10": "F10",
      "F11": "F11",
      "F12": "F12",
      "F2": "F2",
      "F3": "F3",
      "F4": "F4",
      "F5": "F5",
      "F6": "F6",
      "F7": "F7",
      "F8": "F8",
      "F9": "F9",
      "_0": "_0",
      "_1": 0.5,
      "_2": 0.5,
    },
    Object {},
    Object {
      "-": "-",
      "0 ": "0 ",
      "1 ": "1 ",
      "2 ": "2 ",
      "3 ": "3 ",
      "4 ": "4 ",
      "5 ": "5 ",
      "6 ": "6 ",
      "7 ": "7 ",
      "8 ": "8 ",
      "9 ": "9 ",
      "<": "<",
      "=": "=",
      "BACKSPACE": 2,
    },
    Object {
      "A": "A",
      "E": "E",
      "I": "I",
      "O": "O",
      "P": "P",
      "R": "R",
      "T": "T",
      "TAB": 1.5,
      "U": "U",
      "Y": "Y",
      "Z": "Z",
      "[": "[",
      "\\\\": 1.5,
      "]": "]",
    },
    Object {
      "CAPS": 1.75,
      "D": "D",
      "ENTER": 2.25,
      "F": "F",
      "G": "G",
      "H": "H",
      "J": "J",
      "K": "K",
      "L": "L",
      "M": "M",
      "Q": "Q",
      "S": "S",
      "\`": "\`",
    },
    Object {
      ",": ",",
      ":": ":",
      ";": ";",
      "=": "=",
      "B": "B",
      "C": "C",
      "N": "N",
      "SHIFT L": 2.25,
      "SHIFT R": 2.75,
      "V": "V",
      "W": "W",
      "X": "X",
    },
    Object {
      "ALT L": 1.5,
      "ALT R": 1.5,
      "CTRL L": 1.5,
      "CTRL R": 1.5,
      "MENU": "MENU",
      "SPACE": 6,
      "WIN L": "WIN L",
      "WIN R": "WIN R",
    },
  ],
  "German": Array [
    Object {
      "ESC": "ESC",
      "F1": "F1",
      "F10": "F10",
      "F11": "F11",
      "F12": "F12",
      "F2": "F2",
      "F3": "F3",
      "F4": "F4",
      "F5": "F5",
      "F6": "F6",
      "F7": "F7",
      "F8": "F8",
      "F9": "F9",
      "_0": "_0",
      "_1": 0.5,
      "_2": 0.5,
    },
    Object {},
    Object {
      "0 ": "0 ",
      "1 ": "1 ",
      "2 ": "2 ",
      "3 ": "3 ",
      "4 ": "4 ",
      "5 ": "5 ",
      "6 ": "6 ",
      "7 ": "7 ",
      "8 ": "8 ",
      "9 ": "9 ",
      "<": "<",
      "BACKSPACE": 2,
      "´": "´",
      "ß": "ß",
    },
    Object {
      "+": "+",
      "E": "E",
      "I": "I",
      "O": "O",
      "P": "P",
      "Q": "Q",
      "R": "R",
      "T": "T",
      "TAB": 1.5,
      "U": "U",
      "W": "W",
      "Z": "Z",
      "\\\\": 1.5,
      "ü": "ü",
    },
    Object {
      "A": "A",
      "CAPS": 1.75,
      "D": "D",
      "ENTER": 2.25,
      "F": "F",
      "G": "G",
      "H": "H",
      "J": "J",
      "K": "K",
      "L": "L",
      "S": "S",
      "ä": "ä",
      "ö": "ö",
    },
    Object {
      ",": ",",
      "-": "-",
      ".": ".",
      "B": "B",
      "C": "C",
      "M": "M",
      "N": "N",
      "SHIFT L": 2.25,
      "SHIFT R": 2.75,
      "V": "V",
      "X": "X",
      "Y": "Y",
    },
    Object {
      "ALT L": 1.5,
      "ALT R": 1.5,
      "CTRL L": 1.5,
      "CTRL R": 1.5,
      "MENU": "MENU",
      "SPACE": 6,
      "WIN L": "WIN L",
      "WIN R": "WIN R",
    },
  ],
  "USColemak": Array [
    Object {
      "ESC": "ESC",
      "F1": "F1",
      "F10": "F10",
      "F11": "F11",
      "F12": "F12",
      "F2": "F2",
      "F3": "F3",
      "F4": "F4",
      "F5": "F5",
      "F6": "F6",
      "F7": "F7",
      "F8": "F8",
      "F9": "F9",
      "_0": "_0",
      "_1": 0.5,
      "_2": 0.5,
    },
    Object {},
    Object {
      "-": "-",
      "0 ": "0 ",
      "1 ": "1 ",
      "2 ": "2 ",
      "3 ": "3 ",
      "4 ": "4 ",
      "5 ": "5 ",
      "6 ": "6 ",
      "7 ": "7 ",
      "8 ": "8 ",
      "9 ": "9 ",
      "=": "=",
      "BACKSPACE": 2,
      "\`": "\`",
    },
    Object {
      ";": ";",
      "F": "F",
      "G": "G",
      "J": "J",
      "L": "L",
      "P": "P",
      "Q": "Q",
      "TAB": 1.5,
      "U": "U",
      "W": "W",
      "Y": "Y",
      "[": "[",
      "\\\\": 1.5,
      "]": "]",
    },
    Object {
      "'": "'",
      "A": "A",
      "CAPS": 1.75,
      "D": "D",
      "E": "E",
      "ENTER": 2.25,
      "H": "H",
      "I": "I",
      "N": "N",
      "O": "O",
      "R": "R",
      "S": "S",
      "T": "T",
    },
    Object {
      ",": ",",
      ".": ".",
      "/": "/",
      "B": "B",
      "C": "C",
      "K": "K",
      "M": "M",
      "SHIFT L": 2.25,
      "SHIFT R": 2.75,
      "V": "V",
      "X": "X",
      "Z": "Z",
    },
    Object {
      "ALT L": 1.5,
      "ALT R": 1.5,
      "CTRL L": 1.5,
      "CTRL R": 1.5,
      "MENU": "MENU",
      "SPACE": 6,
      "WIN L": "WIN L",
      "WIN R": "WIN R",
    },
  ],
  "USDvorak": Array [
    Object {
      "ESC": "ESC",
      "F1": "F1",
      "F10": "F10",
      "F11": "F11",
      "F12": "F12",
      "F2": "F2",
      "F3": "F3",
      "F4": "F4",
      "F5": "F5",
      "F6": "F6",
      "F7": "F7",
      "F8": "F8",
      "F9": "F9",
      "_0": "_0",
      "_1": 0.5,
      "_2": 0.5,
    },
    Object {},
    Object {
      "0 ": "0 ",
      "1 ": "1 ",
      "2 ": "2 ",
      "3 ": "3 ",
      "4 ": "4 ",
      "5 ": "5 ",
      "6 ": "6 ",
      "7 ": "7 ",
      "8 ": "8 ",
      "9 ": "9 ",
      "BACKSPACE": 2,
      "[": "[",
      "]": "]",
      "\`": "\`",
    },
    Object {
      "'": "'",
      ",": ",",
      ".": ".",
      "/": "/",
      "=": "=",
      "C": "C",
      "F": "F",
      "G": "G",
      "L": "L",
      "P": "P",
      "R": "R",
      "TAB": 1.5,
      "Y": "Y",
      "\\\\": 1.5,
    },
    Object {
      "-": "-",
      "A": "A",
      "CAPS": 1.75,
      "D": "D",
      "E": "E",
      "ENTER": 2.25,
      "H": "H",
      "I": "I",
      "N": "N",
      "O": "O",
      "S": "S",
      "T": "T",
      "U": "U",
    },
    Object {
      ";": ";",
      "B": "B",
      "J": "J",
      "K": "K",
      "M": "M",
      "Q": "Q",
      "SHIFT L": 2.25,
      "SHIFT R": 2.75,
      "V": "V",
      "W": "W",
      "X": "X",
      "Z": "Z",
    },
    Object {
      "ALT L": 1.5,
      "ALT R": 1.5,
      "CTRL L": 1.5,
      "CTRL R": 1.5,
      "MENU": "MENU",
      "SPACE": 6,
      "WIN L": "WIN L",
      "WIN R": "WIN R",
    },
  ],
  "USQwerty": Array [
    Object {
      "ESC": "ESC",
      "F1": "F1",
      "F10": "F10",
      "F11": "F11",
      "F12": "F12",
      "F2": "F2",
      "F3": "F3",
      "F4": "F4",
      "F5": "F5",
      "F6": "F6",
      "F7": "F7",
      "F8": "F8",
      "F9": "F9",
      "_0": "_0",
      "_1": 0.5,
      "_2": 0.5,
    },
    Object {},
    Object {
      "-": "-",
      "0 ": "0 ",
      "1 ": "1 ",
      "2 ": "2 ",
      "3 ": "3 ",
      "4 ": "4 ",
      "5 ": "5 ",
      "6 ": "6 ",
      "7 ": "7 ",
      "8 ": "8 ",
      "9 ": "9 ",
      "=": "=",
      "BACKSPACE": 2,
      "\`": "\`",
    },
    Object {
      "E": "E",
      "I": "I",
      "O": "O",
      "P": "P",
      "Q": "Q",
      "R": "R",
      "T": "T",
      "TAB": 1.5,
      "U": "U",
      "W": "W",
      "Y": "Y",
      "[": "[",
      "\\\\": 1.5,
      "]": "]",
    },
    Object {
      "'": "'",
      ";": ";",
      "A": "A",
      "CAPS": 1.75,
      "D": "D",
      "ENTER": 2.25,
      "F": "F",
      "G": "G",
      "H": "H",
      "J": "J",
      "K": "K",
      "L": "L",
      "S": "S",
    },
    Object {
      ",": ",",
      ".": ".",
      "/": "/",
      "B": "B",
      "C": "C",
      "M": "M",
      "N": "N",
      "SHIFT L": 2.25,
      "SHIFT R": 2.75,
      "V": "V",
      "X": "X",
      "Z": "Z",
    },
    Object {
      "ALT L": 1.5,
      "ALT R": 1.5,
      "CTRL L": 1.5,
      "CTRL R": 1.5,
      "MENU": "MENU",
      "SPACE": 6,
      "WIN L": "WIN L",
      "WIN R": "WIN R",
    },
  ],
}
`;
